client/landing/stepper/declarative-flow/flows/onboarding/onboarding.ts

```diff
@@ -299,10 +299,17 @@ const onboarding: FlowV2< typeof initialize > = {
 			}
 		}, [ currentStepSlug, reduxDispatch, resetOnboardStore ] );
 
-		// Load Survicate
+		/**
+		 * Load Survicate and set visitor traits on each step navigation.
+		 *
+		 * This runs on every step change to ensure:
+		 * - Survicate script loads successfully (retries if initial load failed)
+		 * - Visitor traits are updated when user authentication state changes
+		 * - Analytics tracking works correctly throughout the onboarding flow
+		 */
 		useEffect( () => {
 			addSurvicate();
-		}, [] );
+		}, [ currentStepSlug ] );
```

Comment:
Is changing on step enough, or do we want to trigger it immediately when the user auth state (email for example) changes?